### ISSUE TITLE
Saving screenshot is the wrong format (the one above in the list)

### ISSUE THREAD

**Issue by mellon85** [flameshot-org/flameshot#2270]
### Flameshot Version

Flameshot v11.0.0 (ad1bf28)
Compiled with Qt 5.15.2
darwin: 20.6.0
osx: 11.6

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Big Sur 11.6.2

### Description

When saving a screenshot and I have to select the format it's always off-by-one. Starts with tiff selected and if saved it would save it as PNG, if I select jpeg it will be jp2 instead, etc etc

if I select the topmost (bitmap), it keeps using bitmap.

### Steps to reproduce

1. take any kind of screenshot
2. press on save
3. select a format, it will use the one above the selected 

### System Information

MacOS Big Sur 11.6.2

**Comment by borgmanJeremy** [comment]
Thanks, I can confirm this.

**Comment by AndreaMarangoni** [comment]
I can take this

**Comment by borgmanJeremy** [comment]
Great thanks for the help

**Comment by AndreaMarangoni** [comment]
This looks like a bug on Qt. Even the smallest of the sample has that issue.
The only fix I found is to don't use the native dialogs setting the option `QFileDialog::Option::DontUseNativeDialog`. 
Thoughts?

**Comment by borgmanJeremy** [comment]
If you have a very minimal example can you try building against a newer Qt version? Perhaps even Qt6? I dont really want to use the DontUseNativeDialog option, it makes the Dialog box look terrible on most platforms.

**Comment by AndreaMarangoni** [comment]
I have tried against QT 6.2.2 and it has the same behaviour unfortunately

**Comment by borgmanJeremy** [comment]
I wonder if it's a bug related to auto generating the file formats. In 10.2 we specifically enumerated a few types, but it 11 we auto generate all possible types. #2307 uses the non-native dialogs which we used once before to fix a small bug and got a significant amount of negative feedback.

This is the commit that changed the file format behavior: 3cad8084c73e1a3cbde96a5e0ab8766bdf2bbef5

**Comment by AndreaMarangoni** [comment]
Don't think so. I have created a new project which replicates what we are trying to do in production and tested with different versions of Qt.

This is the sample used.

```
#include <QFileDialog>
#include <QImageWriter>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);
    const QString extension{"png"};
    QString mimeFilter("image/png");
    auto toStringList = []() -> QStringList {
        QStringList mimeList;
        foreach(const auto & mime, QImageWriter::supportedMimeTypes())
        {
            mimeList.push_back(mime);
        }
        return mimeList;
    };
    const QStringList mimeFilters = toStringList();
    const QString file("/Users/andrea/workspace/randomName.png");
    QFileDialog dialog(nullptr, "Choose your image", file);
    dialog.setMimeTypeFilters(mimeFilters);
    dialog.setOptions(QFileDialog::Option::DontUseNativeDialog);
    dialog.setAcceptMode(QFileDialog::AcceptSave);
    dialog.selectMimeTypeFilter(mimeFilter);
    if(QDialog::Accepted == dialog.exec())
    {
    }
    int ret = a.exec();
    return  ret;
}

```

**Comment by borgmanJeremy** [comment]
Was this also a problem in 10.2 then?

**Comment by AndreaMarangoni** [comment]
Completely different dialog..

<img width="886" alt="Screenshot 2022-01-23 at 15 38 45" src="https://user-images.githubusercontent.com/4182715/150686350-0677f8eb-87a3-42e9-a9c7-5e328db3c2ed.png">

**Comment by borgmanJeremy** [comment]
Did you try posting this on the Qt forums? Usually if you post a minimal example to demonstrate the issue they quickly help.

**Comment by AndreaSenseon** [comment]
I am not sure how they can help. I bet they will tell me to use non-native dialog option. It is clear that the behaviour is in Qt 5.x and Qt 6.x. 
The alternatives are:
* use NonNative dialogs (only for Mac) until Qt fixes the issue
* use `QFileDialog::getSaveFileName` generating filters manually from `QImageWriter`
* maybe using a custom QAbstractProxyModel ?

**Comment by AndreaMarangoni** [comment]
> I am not sure how they can help. I bet they will tell me to use non-native dialog option. It is clear that the behaviour is in Qt 5.x and Qt 6.x. The alternatives are:
> 
> * use NonNative dialogs (only for Mac) until Qt fixes the issue
> * use `QFileDialog::getSaveFileName` generating filters manually from `QImageWriter`
> * maybe using a custom QAbstractProxyModel ?

sorry this was me with the wrong account if it wasn't obvious :grimacing:

**Comment by borgmanJeremy** [comment]
Solved with #2379
